Modules/cjkcodecs/multibytecodec.c

```diff
@@ -825,8 +826,13 @@ encoder_encode_stateful(MultibyteStatefulEncoderContext *ctx,
     if (inpos < datalen) {
         if (datalen - inpos > MAXENCPENDING) {
             /* normal codecs can't reach here */
-            PyErr_SetString(PyExc_UnicodeError,
-                            "pending buffer overflow");
+            excobj = PyObject_CallFunction(PyExc_UnicodeEncodeError,
+                                           "ssnns",
+                                           ctx->codec->encoding,
+                                           PyUnicode_AsUTF8(inbuf),
+                                           inpos, datalen,
+                                           "pending buffer overflow");
```

Comment:
missing `if (excobj == NULL) goto errorexit;`

Reply:
Done, for all new and pre-existing code